contrib/config_option_database/utils/MergeYm.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+#############################################################################
+# Copyright (c) 2019 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+from pathlib import Path
+from sys import argv
+
+
+def get_grammar_files():
+    root_dir = Path(__file__).resolve().parents[3]
```

Comment:
This is very sensitive to the module's file path. If it's relocated it won't work.

I see that it's hard to inject a root path, I've looked around in our repository and found that some python functional tests are "cheating": they rely on the `top_builddir` env variable:
`tests/functional/globals.py:    for (root, dirs, files) in os.walk(os.path.abspath(os.path.join(os.environ['top_builddir'], 'modules'))):`

Since we don't install the grammar files, this script has to be used in the source repository.
Just after a quick search, I've found that you can ask the git repository work tree's path with `git rev-parse --show-toplevel`. Maybe this can be injected from outside.

But this can be changed in a subsequent PR.

Reply:
The tarball, we provide as release package is not packed with git folder, so `git rev-parse` might not be the best solution.

Reply:
Emacs has a `locate-dominating-file` function which looks for a specific file name recursively upwards the directory hierarcy, either until the file is found or / directory is reached. Returns with the directory that contains the file name. Similar probing could be implemented, for example you can use `VERSION` or any other file in the project-root directory.

Reply:
Thanks, I will try that!

Reply:
> The tarball, we provide as release package is not packed with git folder, so git rev-parse might not be the best solution.

Since we don't provide the grammar files in the release tarball, you cannot use the script at all in a release tarball, so that does not exclude git rev-parse.

@furiel's idea can work, but it has the possibility that the script may return with the wrong path if it does not found the marker-file (`VERSION` was mentioned).
With a good error-handling (root dir is not allowed), this can be robust.
Maybe the file `syslog-ng-config.h.in` would be a more project-specific file, but there can be additional markers as well: in case the first one (VERSION) is found, it is expected to that the second marker (syslog-ng-config.h.in) is next to it.